modules/javafx.graphics/src/main/native-glass/win/GlassApplication.cpp

```diff
@@ -181,20 +168,25 @@ LRESULT GlassApplication::WindowProc(UINT msg, WPARAM wParam, LPARAM lParam)
             }
             break;
         case WM_SETTINGCHANGE:
+            if (((UINT)wParam == SPI_GETHIGHCONTRAST ||
+                    lParam != NULL && wcscmp(LPCWSTR(lParam), L"ImmersiveColorSet") == 0) &&
+                    m_platformSupport.updatePreferences(m_grefThis)) {
+                return 0;
+            }
             if ((UINT)wParam != SPI_SETWORKAREA) {
                 break;
             }
             // Fall through
         case WM_DISPLAYCHANGE:
             GlassScreen::HandleDisplayChange();
             break;
-        case WM_THEMECHANGED: {
-            JNIEnv* env = GetEnv();
-            jstring themeName = GlassApplication::GetThemeName(env);
-            jboolean result = env->CallBooleanMethod(m_grefThis, javaIDs.Application.notifyThemeChangedMID, themeName);
-            if (CheckAndClearException(env)) return 1;
-            return !result;
-        }
+        case WM_THEMECHANGED:
+        case WM_SYSCOLORCHANGE:
+        case WM_DWMCOLORIZATIONCOLORCHANGED:
+            if (m_platformSupport.updatePreferences(m_grefThis)) {
+                return 0;
+            }
+            break;
```

Comment:
In the case of `WM_THEMECHANGED` we used to always return and not exit the switch statement. Could this cause any problems?

Reply:
Like with `WM_SETTINGCHANGE`, the documentation for [WM_THEMECHANGED](https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-themechanged) states that applications should return zero if they process the message. I think it's good practice to do so, as otherwise Windows wouldn't be aware that we processed the message, causing potential behavioral changes if the default window procedure is changed in future OS releases.